tests/components/hassio/test_repairs.py

```diff
@@ -0,0 +1,389 @@
+    repair_type = "unhealthy" if unhealthy else "unsupported"
+    assert {
+        "breaks_in_ha_version": None,
+        "created": ANY,
+        "dismissed_version": None,
+        "domain": "hassio",
+        "ignored": False,
+        "is_fixable": False,
+        "issue_id": f"{repair_type}_system_{reason}",
+        "issue_domain": None,
+        "learn_more_url": f"https://www.home-assistant.io/more-info/{repair_type}/{reason}",
+        "severity": "critical" if unhealthy else "warning",
+        "translation_key": repair_type,
+        "translation_placeholders": {
+            "reason": reason,
+        },
+    } in issues
+
+
+async def test_unhealthy_repairs(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    hass_ws_client,
+):
+    """Test repairs added for unhealthy systems."""
+    mock_resolution_info(aioclient_mock, unhealthy=["docker", "setup"])
+
+    result = await async_setup_component(hass, "hassio", {})
+    assert result
+
+    client = await hass_ws_client(hass)
+
+    await client.send_json({"id": 1, "type": "repairs/list_issues"})
+    msg = await client.receive_json()
+    assert msg["success"]
+    assert len(msg["result"]["issues"]) == 2
+    assert_repair_in_list(msg["result"]["issues"], unhealthy=True, reason="docker")
+    assert_repair_in_list(msg["result"]["issues"], unhealthy=True, reason="setup")
+
+
+async def test_unsupported_repairs(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    hass_ws_client,
+):
+    """Test repairs added for unsupported systems."""
+    mock_resolution_info(aioclient_mock, unsupported=["content_trust", "os"])
+
+    result = await async_setup_component(hass, "hassio", {})
+    assert result
+
+    client = await hass_ws_client(hass)
+
+    await client.send_json({"id": 1, "type": "repairs/list_issues"})
+    msg = await client.receive_json()
+    assert msg["success"]
+    assert len(msg["result"]["issues"]) == 2
+    assert_repair_in_list(
+        msg["result"]["issues"], unhealthy=False, reason="content_trust"
+    )
+    assert_repair_in_list(msg["result"]["issues"], unhealthy=False, reason="os")
+
+
+async def test_unhealthy_repairs_add_remove(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    hass_ws_client,
+):
+    """Test unhealthy repairs added and removed from dispatches."""
+    mock_resolution_info(aioclient_mock)
+
+    result = await async_setup_component(hass, "hassio", {})
+    assert result
+
+    client = await hass_ws_client(hass)
+
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "supervisor/event",
+            "data": {
+                "event": "health_changed",
+                "data": {
+                    "healthy": False,
+                    "unhealthy_reasons": ["docker"],
+                },
+            },
+        }
+    )
+    msg = await client.receive_json()
+    assert msg["success"]
```

Comment:
Do we need to block till done before listing the repairs to make sure that all tasks are done?

Reply:
I don't think so? It's calling the `@callback` methods to make and delete the issues as part of the task so that should happen right away right? And the task for processing the supervisor event has completed when I receive the json response. So I think checking for success there is effectively accomplisthing a `block_till_done`.

For reference I borrowed this approach for listing the issues from here:
https://github.com/home-assistant/core/blob/b490f47334a92efdfabe205bdc72bb03ad7d2592/tests/components/homeassistant_alerts/test_init.py#L372-L375

Wouldn't that have the same issues if a `block_till_done` was required before listing the issues from the registry?

Reply:
As we create a task for some cases that won't be done before we send the result to the client, there doesn't seem to be a guarantee locally in the integration that the task is done before we receive the result in the client. The ordering in the websocket code may be such though that currently the server tasks will finish before the client receives the result.